### DISCUSSION TITLE
[Idea] Set hardware UUID on arm64 from /etc/machine-id

### DISCUSSION THREAD

**Discussion by yoshi314** [glpi-project/glpi-agent#471] [Ideas]
### Your idea

I've had a problem with original fusioninventory agent - it had no facility to collect hw uuid from rpi4 machine. Because there likely is not one. Not sure how it varies across other arm boards.

I did a workaround by reading custom file with uuid, but since the agent tries to read /etc/machine-id in most cases for various containers - maybe try and use that for this platform?

**Comment by g-bougard** [comment]
Hi @yoshi314 

have you some sample of `/etc/machine-id` content ?

**Comment by yoshi314** [comment]
Well, it's like uuid but without the '-' signs. Same amount of digits/letters.  e.g. e3c1940a72c84b469e48762dad337785

On my installs i patched the agent to read a custom file if uuid was not available and that worked just fine with glpi.

**Reply by g-bougard** [reply]
Okay, so I understand this file contains only one line. So it would be easy to read.

Anyway, I don't see where to start for a patch as you didn't explain on which OS this happens. If you have a patch (even if this is for FI-Agent), you can provide it so I can backport in the best way.

**Reply by yoshi314** [reply]
it happens on arm64 in general. Old FA agent could not find hardware UUID, since it has no codepath for this architecture to do so. 

Dmidecode does not work there. I had debian on r-pi4 at the time.

I can provide a patch against current codebase to achieve the expected result.

**Reply by g-bougard** [reply]
Then, feel free to submit a PR.
